Add cspp

vctt94 opened the pull request:
This PR starts https://github.com/decred/decrediton/issues/2455. We decided to do incremental implementions until we finished this process. 

This PR implements: 

- [x] Add Privacy Page - only allow to see if changed in settings
- [x] Add method to create necessary initial accounts
- [x] Add indicator mixer is running - Still need styling

matheusd commented on app/actions/AccountMixerActions.js:
```diff
@@ -0,0 +1,95 @@
+  // MAINNET DOOR?
+  const walletName = sel.getWalletName(getState());
+  const isTestnet = sel.isTestNet(getState());
+  const csppPort = isTestnet ? "15760" : "";
```
mainnet port is 5760.

matheusd commented on app/actions/DaemonActions.js:
```diff
@@ -346,13 +346,18 @@ export const startWallet = (selectedWallet) => (dispatch, getState) => new Promi
     const lastPoliteiaAccessBlock = walletCfg.get("politeia_last_access_block");
     const dismissBackupRedeemScript = walletCfg.get("dismiss_backup_msg_redeem_script");
+    const enablePrivacy = walletCfg.get("enableprivacy");
+    const mixingAccount = walletCfg.get("mixingaccount");
```
Probably best to follow dcrwallet's convention here and call this the "mixedAccount"

matheusd commented on app/components/views/AccountsPage/Privacy/index.js:
```diff
@@ -0,0 +1,95 @@
+      mixedAccount,
+      changeAccount,
+      mixedAccountBranch,
+      csppServer: "cspp.decred.org:15760"
```
this needs to be parametrized

vctt94 replied:
fixed

matheusd commented on app/config.js:
```diff
@@ -62,6 +66,18 @@ export function initWalletCfg(testnet, walletPath) {
   if (!config.has("ln_macaroonpath")) {
     config.set("ln_macaroonpath", "");
   }
+  if (!config.has("mixingaccount")) {
+    config.set("mixingaccount", "");
```
same as other comment (mixingaccount => mixedaccount)

matheusd commented on app/config.js:
```diff
@@ -74,9 +90,10 @@ export function initWalletCfg(testnet, walletPath) {
 function cleanWalletCfg(config) {
   var key;
   const walletCfgFields = [ "enableticketbuyer", "balancetomaintain", "currency_display",
+    "ln_wallet_exists", "ln_account", "enableprivacy", "mixingaccount", "changeaccount",
```
mixingaccount => mixedaccount

matheusd commented on app/components/views/AccountsPage/Privacy/Page.js:
```diff
@@ -0,0 +1,88 @@
+          <div className="is-row">
+            <div className=""><T id="privacy.mixing.account.branch" m="Account Branch" />:</div>
+            <NumericInput
+              value={mixedAccountBranch} onChange={ e => setMixedAccountBranch(e.target.value)}
```
I think this should be disabled as well, following the convention we're setting of creating the mix config for the user.

Otherwise, at the very least, this needs to be protected against changing from any value besides {0,1}.

vctt94 commented:
Thanks for your review matheus. 

switched all ocurrencies from mixingAccount to mixedAccount, add mixedAccountBranch to the config (I added with the default as 0, I believe we can change it if we see the need).

alexlyp commented:
Looks good overall! 

Can you edit the above PR description that includes what has been added here and a bit more detail?  I can use that as the commit message as well.  Considering this is a large chunk of functionality being added I'd like some extra details in the PR/commit.
